Tests/UtilsTest.swift

```diff
@@ -3,242 +3,269 @@ import XpringKit
 
-    // WHEN it is decoded to an classic address
-    let classicAddressTuple = Utils.decode(xAddress: address)
+        // WHEN the transaction blob is converted to a hash.
+        let transactionHash = Utils.toTransactionHash(transactionBlobHex: transactionBlobHex)
 
-    // THEN the decoded address is undefined.
-    XCTAssertNil(classicAddressTuple)
-  }
+        // THEN the transaction blob is as expected.
+        XCTAssertEqual(
+            transactionHash,
+            "7B9F6E019C2A79857427B4EF968D77D683AC84F5A880830955D7BDF47F120667"
+        )
+    }
+
+    func testToTransactionHashInvalidTransaction() {
+        // GIVEN an invalid transaction blob.
+        let transactionBlobHex = "xrp"
 
-  // MARK: - toTransactionHash
+        // WHEN the transaction blob is converted to a hash.
+        let transactionHash = Utils.toTransactionHash(transactionBlobHex: transactionBlobHex)
 
-  func testToTransactionHashValidTransaction() {
-    // GIVEN a transaction blob.
-    let transactionBlobHex = "120000240000000561400000000000000168400000000000000C73210261BBB9D242440BA38375DAD79B146E559A9DFB99055F7077DA63AE0D643CA0E174473045022100C8BB1CE19DFB1E57CDD60947C5D7F1ACD10851B0F066C28DBAA3592475BC3808022056EEB85CC8CD41F1F1CF635C244943AD43E3CF0CE1E3B7359354AC8A62CF3F488114F8942487EDB0E4FD86190BF8DCB3AF36F608839D83141D10E382F805CD7033CC4582D2458922F0D0ACA6"
+        // THEN the hash is nil.
+        XCTAssertNil(transactionHash)
+    }
 
-    // WHEN the transaction blob is converted to a hash.
-    let transactionHash = Utils.toTransactionHash(transactionBlobHex: transactionBlobHex)
+    // MARK: - toBytesArray
 
-    // THEN the transaction blob is as expected.
-    XCTAssertEqual(
-      transactionHash,
-      "7B9F6E019C2A79857427B4EF968D77D683AC84F5A880830955D7BDF47F120667"
-    )
-  }
+    func testToBytesArrayValidTransactionHash() {
+        // GIVEN a transaction hex.
+        let hex = "9A88C8548E03958FD97AF44AE5A8668896D195A70CF3FF3CB8E57096AA717135"
 
-  func testToTransactionHashInvalidTransaction() {
-    // GIVEN an invalid transaction blob.
-    let transactionBlobHex = "xrp"
+        // WHEN the transaction hex is converted to a bytes array.
+        let hexBytesArray = Utils.toByteArray(hex: hex)
 
-    // WHEN the transaction blob is converted to a hash.
-    let transactionHash = Utils.toTransactionHash(transactionBlobHex: transactionBlobHex)
+        // THEN the transaction hex toHex is as expected.
+        XCTAssertEqual(
+            hexBytesArray.toHex(),
+            "9A88C8548E03958FD97AF44AE5A8668896D195A70CF3FF3CB8E57096AA717135"
+        )
+    }
 
-    // THEN the hash is nil.
-    XCTAssertNil(transactionHash)
-  }
+    func testToBytesArrayInvalidTransactionHash() {
```

Comment:
This test feels odd to me. If we give an invalid input, I'd expect either:
- `nil` output
- an exception to be thrown for an invalid input 

I'd probably prefer to throw an exception rather than making the output optional but happy to hear your thoughts if you have a strong opinion.

I would expect 
```
> Utils.toByteArray(hex: "")
[]
```

There's no way to distinguish this case from the empty input case with the current implementation.